src/System.Runtime.InteropServices.RuntimeInformation/src/RuntimeInformation.Windows.cs

```diff
@@ -1,13 +1,77 @@
 ﻿// Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
+using System.Diagnostics;
+
 namespace System.Runtime.InteropServices
 {
-    public static class RuntimeInformation
+    public static partial class RuntimeInformation
     {
+        private static string s_osDebugName = null;
+        private static Architecture? s_osArch = null;
+
         public static bool IsOSPlatform(OSPlatform osPlatform)
         {
             return OSPlatform.Windows == osPlatform;
         }
+
+        public static string GetOSDebugName()
+        {
+            if (null == s_osDebugName)
+            {
+                string type = Interop.mincore.IsWindowsServer() ? "Server" : "Client";
+                string version = "Unknown";
+
+                if (Interop.mincore.IsWindows10OrGreater())
+                    version = "10";
```

Comment:
On my Win10 machine, in a .NET 4.6 console app, the code in this PR tells me that I'm on Win 8 or greater, so apparently VerifyVersionInfo isn't behaving the way we'd want it to.  I assume if I were to add a manifest claiming the right OSes than it would work.  But at that point, I believe GetVersion would work as well.  I'm just not seeing why we should be in the business of doing anything more than asking the OS for its version and giving that back, whatever it tells it... if it chooses to lie, that's its problem, we're just the messenger.  We can't do any better than what the OS does... let's not do worse.

In terms of other functions, on my machine, the following prints out "Windows 10.0.10240":
```C#
private unsafe struct OSVERSIONINFOW
{
    internal int dwOSVersionInfoSize;
    internal int dwMajorVersion;
    internal int dwMinorVersion;
    internal int dwBuildNumber;
    internal int dwPlatformId;
    internal fixed char szCSDVersion[128];
}

[DllImport("ntdll", SetLastError = true)]
private static extern int RtlGetVersion(ref OSVERSIONINFOW lpVersionINformation);

internal static unsafe string GetDisplayName()
{
    var v = new OSVERSIONINFOW() { dwOSVersionInfoSize = Marshal.SizeOf<OSVERSIONINFOW>() };
    RtlGetVersion(ref v);
    return $"Windows {v.dwMajorVersion}.{v.dwMinorVersion}.{v.dwBuildNumber} {new string(v.szCSDVersion)}";
}
```

On my machine, shelling out to "ver" gives back:
```
Microsoft Windows [Version 10.0.10240]
```

There are probably others, too.